File: api/src/domain/enums.ts
export enum BillingType {
  SALARY = 'SALARY',
  AWARD = 'AWARD',
}

export enum ExpenseType {
  FIXED = 'FIXED',
  CARD = 'CARD',
  DETACHED = 'DETACHED',
}

export enum ExpenseStatus {
  PAID = 'PAID',
  PENDING = 'PENDING',
}

// Conjuntos (Sets) para facilitar a validação de regras de negócio
export const RECURRING_BILLING_TYPES = new Set<string>([BillingType.SALARY]);
export const RECURRING_EXPENSE_TYPES = new Set<string>([
  ExpenseType.FIXED,
  ExpenseType.CARD,
]);

File: backend/domain/enums.py
from enum import Enum


class BillingType(str, Enum):
    SALARY = "SALARY"
    AWARD = "AWARD"


class ExpenseType(str, Enum):
    FIXED = "FIXED"
    CARD = "CARD"
    DETACHED = "DETACHED"


class ExpenseStatus(str, Enum):
    PAID = "PAID"
    PENDING = "PENDING"


# Which types are automatically recurring per month
RECURRING_BILLING_TYPES = {BillingType.SALARY}
RECURRING_EXPENSE_TYPES = {ExpenseType.FIXED, ExpenseType.CARD}
